setup_build.py

```diff
@@ -130,7 +131,7 @@ def run(self):
         import numpy
 
         complex256_support = hasattr(numpy, 'complex256') and \
-            os.environ.get('CIBW_ARCHS_MACOS') != 'arm64'
+            (sys.platform != "darwin" or platform.processor() != "arm")
```

Comment:
This seems too general, if numpy has float128 support on a system, then this should work (It would be good to work out why MacOS doesn't give the correct result already).

Reply:
It looks like this comes from https://github.com/h5py/h5py/pull/2065, which shouldn't be needed if we're not cross-compiling.